### PR TITLE
Ignore forced running splits when checking minimumNumberOfDrivers.

### PR DESCRIPTION
**Pull request by pnowojski**
Otherwise with (for example) minimumNumberOfDrivers = 100, 200 "forceStart" splits
and 100 "normal" splits, depending on order of appearing splits, number of simultaneously running splits may vary. If normal splits start first, there will be 300 running splits. If "forceStart" splits start first, there will be only 200 running splits.

This problem was observed on tpch q21 query:

```
set session join_distribution_type = 'automatic';

SELECT s.name, count(*) as numwait FROM "supplier" s, "lineitem" l1, "orders" o, "nation" n WHERE s.suppkey = l1.suppkey AND o.orderkey = l1.orderkey AND o.orderstatus = 'F' AND l1.receiptdate> l1.commitdate AND EXISTS ( SELECT * FROM "lineitem" l2 WHERE l2.orderkey = l1.orderkey AND l2.suppkey <> l1.suppkey ) AND NOT EXISTS ( SELECT * FROM "lineitem" l3 WHERE l3.orderkey = l1.orderkey AND l3.suppkey <> l1.suppkey AND l3.receiptdate > l3.commitdate ) AND s.nationkey = n.nationkey AND n.name = 'SAUDI ARABIA' GROUP BY s.name ORDER BY numwait DESC, s.name LIMIT 1;
```

With `set session join_distribution_type = 'repartitioned';` (old `distributed_joins=true`) this query completed in ~42 seconds against tpch 100gb on 8 node cluster. With `set session join_distribution_type = 'automatic';` it completed randomly between 30 seconds and 60 seconds WITHOUT this PR. With this PR it has stable execution times 27-29seconds.

code base: https://github.com/Teradata/presto/tree/sprint-44

Link to explain for the above query: http://pastebin.com/DEiy65pb

Problem was with race between splits from fragment 3 (probe side) and build splits from fragment 2.

### CODE REVIEW

**Comment by dain** [comment]
Great find.  I actually thought we were already doing this.

**Comment by pnowojski** [comment]
@dain sure. Added and rebased on latest master.

File: presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java [hunk 1]
```diff
@@ -750,6 +771,12 @@ public synchronized int getTotalSplits()
     }
 
     @Managed
+    public int getForcedRunningSplits()
+    {
+        return forcedRunningSplits.size();
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java]: Unsynchronized access to non-thread-safe object.

Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java]: fixed

**Comment by pnowojski** [comment]
Thanks!

**Pull request merged**